using_images/other_images/jenkins_slaves.adoc

```diff
@@ -236,3 +237,44 @@ variables.
 * Set the maximum heap size and JVM arguments for any Gradle test JVM by
 the maxHeapSize and jvmArgs settings in build.gradle, or though the
 `-Dorg.gradle.jvmargs` command line argument.
+
+[[slave-pod-retention]]
+== Slave Pod Retention
+
+Jenkins slave pods are deleted by default after the build completes or is aborted. 
+This behavior can be changed via the Kubernetes plug-in "Pod Retention" setting. 
+Pod retention can be set for all Jenkins builds, with overrides for each 
+pod template. The following behaviors are supported:
+
+- "Always" will keep the build pod regardless of build result.
+- "Default" will use the plug-in value (pod template only).
+- "Never" always deletes the pod.
+- "On Failure" keeps the pod if it fails during the build.
+
+Pod retention can also be overrode in the pipeline Jenkinsfile as follows:
+
+====
+[source,groovy]
+----
+podTemplate(label: "mypod",
+  cloud: "openshift",
+  inheritFrom: "maven",
+  podRetention: onFailure(), <1>
+  containers: [
+    ...
+  ]) {
+  node("mypod") {
+    ...
+  }
+}
+----
+<1> Allowed values for `podRetention` are `never()`, `onFailure()`, `always()`, and `default()`.
+====
+
+[WARNING]
+====
+Pods that are kept may continue to run and count against resource quotas.
+====
+
+See the link:https://github.com/jenkinsci/kubernetes-plugin[Kubernetes plug-in
```

Comment:
ifdef::openshift-origin[]
For more information, see the link:https://github.com/jenkinsci/kubernetes-plugin[Kubernetes plug-in documentation].
endif::openshift-origin[]